src/model/transaction/splitBlockInContentState.js

```diff
@@ -46,6 +46,7 @@ function splitBlockInContentState(
     key: keyBelow,
     text: text.slice(offset),
     characterList: chars.slice(offset),
+    data: Object(),
```

Comment:
That seems a little overkill. The callback would also have to be threaded all the way to this function, and it wouldn't match any other transaction or `Modifier` functions.

My instinct here is that the data should be preserved for both blocks, then the client can manually remove the data if it's not desired for the specific use case. It means performing an extra step, but there's no magic or configuration involved.